sklearn/model_selection/tests/test_search.py

```diff
@@ -1750,6 +1752,42 @@ def get_n_splits(self, *args, **kw):
         ridge.fit(X[:train_size], y[:train_size])
 
 
+@pytest.mark.parametrize(
+    "return_train_score, expected_msgs",
+    [(False, ('One or more of the test scores are non-finite')),
+     (True, ("One or more of the test scores are non-finite",
+             "One or more of the train scores are non-finite"))]
+)
+def test_gridsearchcv_raise_warning_with_non_finite_score(
+        return_train_score, expected_msgs):
+    # Non-regression test for:
+    # https://github.com/scikit-learn/scikit-learn/issues/10529
+    # Check that we raise a UserWarning when a non-finite score is
+    # computed in the GridSearchCV
+    X = norm(-1, 0.5).rvs(100, random_state=np.random.RandomState(28))
+    kernel = 'epanechnikov'
+    steps = 10
+    lower = 0.0194867441113
+    upper = 0.0974337205567
+    bandwidth_range = np.linspace(lower, upper, steps)
+    grid = GridSearchCV(
+        KernelDensity(kernel=kernel),
+        param_grid={'bandwidth': bandwidth_range},
+        cv=20,
+        return_train_score=return_train_score
+    )
+
+    with pytest.warns(UserWarning) as warnings:
+        grid.fit(X[:, np.newaxis])
+
+    warnings = list(map(lambda warning: str(warning.message), warnings))
+    warnings = ",".join(warnings)
+    assert expected_msgs[0] in warnings
+
+    if return_train_score:
+        assert expected_msgs[1] in warnings
+
+
```

Comment:
I think that this test is better because we are handling the way the failure happens using our own scorer.
The issue that you had was that your test was creating `nan` during the test but not during the test. Now because we are mocking the scorer we can issue non-finite value whenever we want.